Intersect.Server/Entities/Player.cs

```diff
@@ -4619,6 +4621,32 @@ public void EquipmentProcessItemLoss(int slot)
             PacketSender.SendEntityStats(this);
         }
 
+        /// <summary>
+        /// Unequips any items that the player is currently wearing in which they no longer meet the conditions for
+        /// Also unequips any items that are not actually equipment anymore
+        /// </summary>
+        public void UnequipInvalidItems()
+        {
+            var updated = false;
+            for (int i = 0; i < Options.EquipmentSlots.Count; i++)
+            {
+                var itemSlot = Equipment[i];
+                if (itemSlot > -1)
+                {
+                    if (Items[itemSlot] == null || Items[itemSlot].ItemId == Guid.Empty || Items[itemSlot].Descriptor == null || Items[itemSlot].Descriptor.ItemType != ItemTypes.Equipment || !Conditions.MeetsConditionLists(Items[itemSlot].Descriptor.UsageRequirements, this, null)) {
```

Comment:
Please break this check up into parts. This is far too long and unreadable.

I suggest something like:
```csharp
var item = Items[itemSlot];
var descriptor = item?.Descriptor;
if (descriptor != default && item.ItemId != Guid.Empty) {
    continue;
}

if (descriptor.ItemType == ItemTypes.Equipment) {
    continue;
}

if (Conditions.MeetsConditionLists(descriptor.UsageRequirements, this, null)) {
    continue;
}

Equipment[i] = -1;
updated = true;
```